dashboard/app/api.go

```diff
@@ -139,26 +139,23 @@ func handleAPI(c context.Context, r *http.Request) (reply interface{}, err error
 	password := r.PostFormValue("key")
 	ns, err := checkClient(getConfig(c), client, password, subj)
 	if err != nil {
-		if client != "" {
-			log.Errorf(c, "checkClient('%s') error: %v", client, err)
-		} else {
-			// Don't log as error if somebody just invokes /api.
-			log.Infof(c, "checkClient('%s') error: %v", client, err)
-		}
-		return nil, err
+		return nil, fmt.Errorf("checkClient('%s') error: %w", client, err)
 	}
 	var payload []byte
 	if str := r.PostFormValue("payload"); str != "" {
+		ungzipErr := func(err error) error {
+			return fmt.Errorf("failed to ungzip payload: %v: %w", err.Error(), ErrClientBadRequest)
```

Comment:
All errors from authorized clients should be logged.
While it's "bad client request", the client is ours and we need to know if our clients are sending bad requests.
ErrClientBadRequest is intended for web requests, where users type nonsense into url.

Reply:
Yes, it is client error. But ErrClientBadRequest does not mean if it's client error or not. It means log level to use.
These errors should be logged.
If ErrClientBadRequest is confusing, you may rename it.

Reply:
Moved to errors.